### PR TITLE
Update Form.php

### PR DESCRIPTION
**Pull request by andrii-pukhalevych**
public function Form::set() - Saves a variable or an associative array of variables for use inside form data

Allows to set single element inside form data, similar to Session::write() or View::set()

### CODE REVIEW

**Comment by andrii-pukhalevych** [comment]
Please create tests if PR is acceptable. I didnt work with phpunit

**Comment by ADmad** [comment]
What's your use case which requires setting individual values?  Can't you just aggregate all data beforehand and use `Form::setData()`?

**Comment by andrii-pukhalevych** [comment]
In my form default values depends on user type, so code looks more clean when data items set separately, without aggregating data into additional variable.

**Comment by ADmad** [comment]
You can still have separate variables for common and user specific data and then merge only when making th `setData()` call.

**Comment by andrii-pukhalevych** [comment]
Its more clean to write shorter and without additional variable, like using Controller::set()
```
$Form->setData(['var1'=>'val1']);
if ($condition)
   $Form->set('var2', 'val2');
```

Instead of this:
```
$data = ['var1'=>'val1'];
if ($condition)
    $data['var2'] = 'val2';
$Form->setData($data);
```

File: src/Form/Form.php [hunk 1]
```diff
@@ -270,6 +270,28 @@ public function getData(?string $field = null)
...
 
+    /**
+     * Saves a variable or an associative array of variables for use inside form data.
+     *
+     * @param string|array $name A string or an array of data.
+     * @param mixed $value Value in case $name is a string (which then works as the key).
+     *   Unused if $name is an associative array, otherwise serves as the values to $name's keys.
+     * @return $this
+     */
+	public function set($name, $value = null)
```
Comment (markstory) [inline src/Form/Form.php]: If we're going to add a new method, could it only add a single key? We already have `setData()` to set all the keys.

Comment (andrii-pukhalevych) [reply src/Form/Form.php]: `setData()` overrides all existing data. `set()` appends new data using `Hash::insert()`.

		$form->set('var1', 'val1'); //array ( 'var1' => 'val1', )
		$form->set(['var2'=>'val2', 'var3'=>'val3']); //array ( 'var1' => 'val1', 'var2' => 'val2', 'var3' => 'val3', )
		$form->set(['var1'=>'val1_new']); //array ( 'var1' => 'val1_new', 'var2' => 'val2', 'var3' => 'val3', )
		$form->setData(['var1'=>'val1_new2']); //array ( 'var1' => 'val1_new2', )

Comment (markstory) [reply src/Form/Form.php]: Ok. Are you aiming to replicate the behavior of `ViewVarsTrait::set()` on Form?

Comment (andrii-pukhalevych) [reply src/Form/Form.php]: Yes. Actually I want to replicate `EntityTrait::set()` first, and `EntityTrait::setError()` in next PR.
Both are used as forms context, but set() and setError() works for entity only.

Comment (markstory) [reply src/Form/Form.php]: >  Both are used as forms context, but set() and setError() works for entity only.

I don't see `set()` and `setError()` being used in the `FormContext`?

Comment (andrii-pukhalevych) [reply src/Form/Form.php]: I mean that entities and forms are used as form context. Not methods.

**Comment by andrii-pukhalevych** [comment]
@markstory Did you completely decline this PR?

**Comment by saeideng** [comment]
@andrii-pukhalevych branch `4.x` has been deleted so PR(s) on it closed automatically 
current branch : 
`master` => for version 4.0.x
`4.next` =>for 4.1.x

**Comment by saeideng** [comment]
you can open a new Issue to discussion

**Comment by markstory** [comment]
I didn't mean for this to get closed, I was removing the now stale 4.x branch. If you could open the pull request against master that would be appreciated as I cannot change the base branch now.

**Pull request closed without merging**